File: 2789-largest-element-in-an-array-after-merge-operations/2789-largest-element-in-an-array-after-merge-operations.py
class Solution:
    def maxArrayValue(self, nums: List[int]) -> int:
        n = len(nums)
        result = total = nums[n - 1]

        for i in range(n - 2, -1, -1):
            if nums[i] <= total:
                total += nums[i]
            else:
                total = nums[i]

            result = max(result, total)

        return result

File: 2789-largest-element-in-an-array-after-merge-operations/2789-largest-element-in-an-array-after-merge-operations.ts
function maxArrayValue(nums: number[]): number {
    const n = nums.length;
    let result = nums[n - 1];
    let total = nums[n - 1];

    for (let i = n - 2; i >= 0; i--) {
        if (nums[i] <= total) {
            total += nums[i];
        } else {
            total = nums[i];
        }

        result = Math.max(result, total);
    }

    return result;
};